app/data_providers/dispatch.py

```diff
@@ -51,25 +52,29 @@ async def after_save(self, resolved_id: int, count: int) -> List[CorpusRecommend
         except Item2ItemError:
             # do not fallback for "Similar stores" after saving
             recs = []
-        return self._sample(recs, count)
+        return self._to_corpus_items(recs, count)
 
     @_empty_on_error
     async def after_article(self, resolved_id: int, count: int) -> List[CorpusRecommendationModel]:
         try:
-            recs = await self.item_recommender.related(resolved_id, count)
+            recs = await self.item_recommender.related(resolved_id, 20)
```

Comment:
it is not guaranteed that there will be 5 different domains in those 20 items, but let's start with this and increase it later if we still see a lot of duplicates after testing